### ISSUE TITLE
[Bug]: Rotated tiles are not displayed correctly + collision for rotated tiles is bugged

### ISSUE THREAD

**Issue by dvmatyun** [RafaelBarbosatec/bonfire#531]
### What happened?

Using "Z" key to rotate a tile in tiled editor results in wrong placement in flutter.

### Steps to reproduce?

```Markdown
1. Rotate a tile with collision in Tiled editor
2. Run a map using an example
3. Tile placed incorrectly
```

### What did you expect to happen?

1. I expected tile to be placed correctly
![image](https://github.com/RafaelBarbosatec/bonfire/assets/47291701/7982cd17-cc4f-4992-883e-8186114572b4)
2. Actual in-game:
![image](https://github.com/RafaelBarbosatec/bonfire/assets/47291701/ccda5df0-b2b7-4305-9196-bf9b89b7fa46)

### Bonfire version

version: 3.9.5

**Comment by dvmatyun** [comment]
PS working fine if you rotate using "X" key in tiled editor

**Comment by dvmatyun** [comment]
> PS working fine if you rotate using "X" key in tiled editor
BUT if you rotate a tile using X key and it has collision - the Knight will be stuck in that wall by the collision. Does not happen with same tile that was not rotated by "X" key. (Collision was set by pressing the key that makes a simple rectangle collision for tile, see screenshot)
![image](https://github.com/RafaelBarbosatec/bonfire/assets/47291701/58e702d4-e678-4466-ba0d-e4a64ed045f2)

**Comment by RafaelBarbosatec** [comment]
Hi @dvmatyun !
I will investigate the tile rotate. The collision rotate it was not implemented yet.
Any contribution is welcome!

**Comment by RafaelBarbosatec** [comment]
Fix available in version `3.9.6`
